### PR TITLE
arm64: dts: qcom: msm8916-asus-z010d: initial device tree

### PR DESCRIPTION
**Pull request by devajithvs**
Initial device tree for Asus Zenfone Max ZC550KL

### CODE REVIEW

File: arch/arm64/boot/dts/qcom/msm8916-asus-z010d.dts [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&gpio_keys_default>;
+
+		label = "GPIO Buttons";
+
+		volume-up {
+			label = "Volume Up";
+			gpios = <&msmgpio 107 GPIO_ACTIVE_LOW>;
+			linux,input-type = <1>;
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-asus-z010d.dts:38]: unneeded, this is the default

File: arch/arm64/boot/dts/qcom/msm8916-asus-z010d.dts [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+			linux,code = <KEY_VOLUMEUP>;
+			gpio-key,wakeup;
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-asus-z010d.dts:40]: I don't think this exists on mainline, also wakeup on volume keys doesn't really make sense.

File: arch/arm64/boot/dts/qcom/msm8916-asus-z010d.dts [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+	};
+
+	l9 {
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+	};
+
+	l10 {
+		regulator-min-microvolt = <3000000>;
+		regulator-max-microvolt = <3000000>;
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-asus-z010d.dts:144]: Where does the 3V come from here?

File: arch/arm64/boot/dts/qcom/msm8916-asus-z010d.dts [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+	};
+
+	usb_id_default: usb-id-default {
+		pins = "gpio110";
+		function = "gpio";
+
+		drive-strength = <8>;
+		bias-pull-up;
+	};
+};
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-asus-z010d.dts:206]: add new line at end of file please :)

File: arch/arm64/boot/dts/qcom/msm8916-asus-z010d.dts [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+
+	l18 {
+		regulator-min-microvolt = <2700000>;
+		regulator-max-microvolt = <2700000>;
+	};
+};
+
+&msmgpio {
+	gpio_keys_default: gpio-keys-default {
+		pins = "gpio107", "gpio108", "gpio109";
```
Comment (Minecrell) [inline arch/arm64/boot/dts/qcom/msm8916-asus-z010d.dts:192]: Why do you configure 3 GPIOs here, it seems like only GPIO107 is used.

File: arch/arm64/boot/dts/qcom/msm8916-asus-z010d.dts (hunk 3 above)
Comment (devajithvs) [reply arch/arm64/boot/dts/qcom/msm8916-asus-z010d.dts:144]: From [here](https://github.com/kernel-hut/android_kernel_asus_msm8916/blob/1d3b276bcd518d8850f68b0a4e3561f207a5e1fb/arch/arm64/boot/dts/qcom/msm8916-regulator.dtsi). Line 298.
```
	rpm-regulator-ldoa10 {
		status = "okay";
		pm8916_l10: regulator-l10 {
			regulator-min-microvolt = <3000000>;
			regulator-max-microvolt = <3000000>;
			qcom,init-voltage = <3000000>;
			status = "okay";
		};
	};
```

Comment (Minecrell) [reply arch/arm64/boot/dts/qcom/msm8916-asus-z010d.dts:144]: but isn't that the wrong source?

https://github.com/kernel-hut/android_kernel_asus_msm8916/blob/1d3b276bcd518d8850f68b0a4e3561f207a5e1fb/arch/arm64/boot/dts/qcom/zc550kl_msm8916-regulator.dtsi#L297-L305

Comment (devajithvs) [reply arch/arm64/boot/dts/qcom/msm8916-asus-z010d.dts:144]: I just cross checked with the de-compiled dtc. It seems to be a bit different there.
```
rpm-regulator-ldoa10 {
				compatible = "qcom,rpm-smd-regulator-resource";
				qcom,resource-name = "ldoa";
				qcom,resource-id = <0xa>;
				qcom,regulator-type = <0x0>;
				qcom,hpm-min-load = <0x2710>;
				status = "okay";

				regulator-l10 {
					compatible = "qcom,rpm-smd-regulator";
					regulator-name = "8916_l10";
					qcom,set = <0x3>;
					status = "okay";
					regulator-min-microvolt = <0x2932e0>;
					regulator-max-microvolt = <0x2ab980>;
					qcom,init-voltage = <0x2932e0>;
					linux,phandle = <0x49>;
					phandle = <0x49>;
				};
			};
```
That translates to:
regulator-min-microvolt = 2700000;
regulator-max-microvolt = 2800000;

I'm lucky to have not fried anything :)

Comment (Minecrell) [reply arch/arm64/boot/dts/qcom/msm8916-asus-z010d.dts:144]: It's somewhat safe until you make use of the regulator. But make sure to look at the correct device tree sources or the decompiled dtb :)

File: arch/arm64/boot/dts/qcom/msm8916-asus-z010d.dts (hunk 4 above)
Comment (devajithvs) [reply arch/arm64/boot/dts/qcom/msm8916-asus-z010d.dts:206]: Will do :)

File: arch/arm64/boot/dts/qcom/msm8916-asus-z010d.dts (hunk 1 above)
Comment (devajithvs) [reply arch/arm64/boot/dts/qcom/msm8916-asus-z010d.dts:38]: Can I completely remove the volume-up block?

Comment (Minecrell) [reply arch/arm64/boot/dts/qcom/msm8916-asus-z010d.dts:38]: Well, the volume-up button might be useful :)

File: arch/arm64/boot/dts/qcom/msm8916-asus-z010d.dts (hunk 3 above)
Comment (devajithvs) [reply arch/arm64/boot/dts/qcom/msm8916-asus-z010d.dts:144]: I think I might have to redo the voltages from the decompiled dtc.

**Comment by devajithvs** [comment]
I guess, that resolves everything. I'll update here after I flash this on my device.

**Comment by devajithvs** [comment]
Boots fine. 
wifi, usb, sdcard - all working.

**Review by Minecrell — APPROVED** [review]

**Pull request merged**